qiskit/providers/backend.py

```diff
@@ -172,13 +172,24 @@ def status(self):
         Returns:
             BackendStatus: the status of the backend.
         """
-        return BackendStatus(
-            backend_name=self.name(),
-            backend_version="1",
-            operational=True,
-            pending_jobs=0,
-            status_msg="",
-        )
+        with warnings.catch_warnings():
+            # BackendV1 is deprecated along qiskit.providers.models.backendstatus.BackendStatus
+            # They both need to be removed at the same time
+            warnings.filterwarnings(
+                "ignore",
+                category=DeprecationWarning,
+                message=r"qiskit\.providers\.models.+",
+                module="qiskit",
+            )
+            from qiskit.providers.models.backendstatus import BackendStatus
+
+            return BackendStatus(
+                backend_name=self.name(),
+                backend_version="1",
+                operational=True,
+                pending_jobs=0,
+                status_msg="",
+            )
```

Comment:
I personally wouldn't put this warning catch in - it's expected that deprecated code will call other deprecated code, and the `stacklevel` of the `BackendStatus` warning should be set such that `qiskit` is "blamed" for the warning, which will mean it's suppressed for users by default (which is good - we want that).  It would show up if all warnings are set to display, but that's expected behaviour.